demo_interface.py

```diff
@@ -0,0 +1,401 @@
+            dcc.Store(id="selected-features", data=[]),
+            dcc.Store(id="soln-score", data=0.0),
+            # Header brand banner
+            html.Div(className="banner", children=[html.Img(src=THUMBNAIL)]),
+            # Settings and results columns
+            html.Div(
+                className="columns-main",
+                children=[
+                    # Left column
+                    html.Div(
+                        id={"type": "to-collapse-class", "index": 0},
+                        className="left-column",
+                        children=[
+                            html.Div(
+                                className="left-column-layer-1",  # Fixed width Div to collapse
+                                children=[
+                                    html.Div(
+                                        className="left-column-layer-2",  # Padding and content wrapper
+                                        children=[
+                                            html.H1(MAIN_HEADER),
+                                            html.P(DESCRIPTION),
+                                            generate_settings_form(),
+                                            generate_run_buttons(),
+                                        ],
+                                    )
+                                ],
+                            ),
+                            # Left column collapse button
+                            html.Div(
+                                html.Button(
+                                    id={"type": "collapse-trigger", "index": 0},
+                                    className="left-column-collapse",
+                                    children=[html.Div(className="collapse-arrow")],
+                                ),
+                            ),
+                        ],
+                    ),
+                    # Right column
+                    html.Div(
+                        className="right-column",
+                        children=[
+                            dcc.Tabs(
+                                id="tabs",
+                                value="input-tab",
+                                mobile_breakpoint=0,
+                                children=[
+                                    dcc.Tab(
+                                        label="Input",
+                                        id="input-tab",
+                                        value="input-tab",  # used for switching tabs programatically
+                                        className="tab",
+                                        children=[
+                                            dcc.Loading(
+                                                parent_className="input",
+                                                type="circle",
+                                                color=THEME_COLOR_SECONDARY,
```

Comment:
dcc.Loading has a delay feature which will address the flicker by delaying the responsiveness of the loading to however many milliseconds you want. You can play around with how many ms is good, I tried with 50 and it seemed to fix it.

```suggestion
                                                color=THEME_COLOR_SECONDARY,
                                                delay_show=50,
```